packages/typespec-vscode/src/emit/emit.ts

```diff
@@ -0,0 +1,335 @@
+    };
+  };
+
+  const registerEmitters = getRegisterEmitters(kind);
+  const all = [...registerEmitters].map((e) => toQuickPickItem(e));
+
+  const selectedEmitter = await vscode.window.showQuickPick<EmitQuickPickItem>(all, {
+    title: "Select a Language",
+    canPickMany: false,
+    placeHolder: "Pick a Language",
+    ignoreFocusOut: true,
+  });
+
+  if (!selectedEmitter) {
+    logger.info("No emitter selected. Generating Cancelled.", [], {
+      showOutput: false,
+      showPopup: false,
+    });
+    return;
+  }
+
+  logger.info("npm install...", [], {
+    showOutput: false,
+    showPopup: false,
+  });
+
+  const npmUtil = new NpmUtil(baseDir);
+  const packagesToInstall: string[] = [];
+
+  /* install emitter package. */
+  logger.info(`select ${selectedEmitter.package}`);
+  const { action, version } = await npmUtil.ensureNpmPackageInstallAction(
+    selectedEmitter.package,
+    selectedEmitter.version,
+  );
+
+  if (action === InstallationAction.Upgrade) {
+    logger.info(`Upgrading ${selectedEmitter.package} to version ${version}`);
+    const options = {
+      ok: `OK (install ${selectedEmitter.package}@${version} by 'npm install'`,
+      recheck: `Check again (install ${selectedEmitter.package} manually)`,
+      ignore: `Ignore (don't upgrade emitter ${selectedEmitter.package})`,
+    };
+    const selected = await vscode.window.showQuickPick(Object.values(options), {
+      canPickMany: false,
+      ignoreFocusOut: true,
+      placeHolder: `Package '${selectedEmitter.package}' needs to be upgraded for generating`,
+      title: `TypeSpec Generating...`,
+    });
+    if (selected === options.ok) {
+      packagesToInstall.push(`${selectedEmitter.package}@${version}`);
+    } else if (selected === options.ignore) {
+      logger.info(`Ignore upgrading emitter ${selectedEmitter.package} for generating`, [], {
+        showOutput: false,
+        showPopup: false,
+      });
+    } else {
+      logger.info(
+        `Need to manually install the package ${selectedEmitter.package}@${version}. Generating Cancelled.`,
+        [],
+        {
+          showOutput: false,
+          showPopup: true,
+        },
+      );
+      return;
+    }
+  } else if (action === InstallationAction.Install) {
+    let packageFullName = selectedEmitter.package;
+    if (version) {
+      packageFullName = `${selectedEmitter.package}@${version}`;
+    }
+    logger.info(`Installing ${packageFullName}`);
+    /* verify dependency packages. */
+    const dependenciesToInstall = await npmUtil.ensureNpmPackageDependencyToUpgrade(
+      selectedEmitter.package,
+      version,
+      npmDependencyType.peerDependencies,
+    );
+    logger.info(`${dependenciesToInstall}`);
+    if (dependenciesToInstall.length > 0) {
+      vscode.window.showInformationMessage(
+        `Need to manually upgrade following dependency packages: ${dependenciesToInstall.join("\\n")}. \nGenerating Cancelled`,
```

Comment:
no option to install them for user?

Reply:
For dependency packages, we have not design the screen, for current decision, we will not auto-install this for customer, and just inform customer and cancel generation. May update after design ready.